src/object/authenticate.c

```diff
@@ -6150,7 +6150,7 @@ au_fetch_class_internal (MOP op, SM_CLASS ** class_ptr, AU_FETCHMODE fetchmode,
 	}
     }
 
-  if (Au_disable || !(error = check_authorization (classmop, class_, type)))
+  if ((Au_disable && type != DB_AUTH_ALTER) || !(error = check_authorization (classmop, class_, type)))
```

Comment:
Here is the code for the au_fetch_instance()function.

```
  if (Au_disable || !(error = check_authorization (classmop, class_, type)))
    {
      error = fetch_instance (op, obj_ptr, mode, fetch_version_type);
    }
```
    
Don't you need to fix it in the same context as the edit here?

Reply:
au_fetch_instance is not called at ALTERing a table or a view. Authenticate check needs only at ALTERing.

Reply:
How about below for readability?
```
au_fetch_class_internal ()
if ((Au_disable && !Au_sysadm) || !(error = check_authorization (classmop, class_, type)))

check_authorization ()
if (Au_disable && !(type == DB_AUTH_ALTER && (sm_class->flags & SM_CLASSFLAG_SYSTEM)) )
```

Reply:
Ok, thanks